### ISSUE TITLE
Libation on Mac crashing when download completing

### ISSUE THREAD

**Issue by VicGar87** [rmcrackan/Libation#918]
**Describe the bug**
When Libation downloads a book, the download finishes, but then the program crashes. This behavior started approximately one month ago. Prior to that, everything was running smoothly. I was using Libation Chardonnay 11.1.0 x64 at the time, then when this issue began, I tried several things, including updating to 11.3.5.

**To Reproduce**
1. I click on a "traffic light" next to my chosen book to "liberate" it.
2. The audiobook shows in the process queue and the progress bar indicates download progress
3. The download progress bar completes, then the program closes and the operating system window "Problem Report for Libation" appears with a report in the window.
4. Upon reopening Libation, I find that the audiobook has indeed been downloaded and is in the specified destination folder, but the program still indicates that it is "Not Liberated"

**Expected behavior**
The program to not crash and be able to download more audiobooks without needing to reopen the program after every download.

**What I've attempted to fix the issue with no change in behavior**
1.  Deleted AccountSettings.json and relogged into Audible account.
2. Changed the temporary file location from the default Temporary folder to my Users folder.
3. Changed the destination folder from a network drive to a local drive.
4. Updated the program to Libation Chardonnay 11.3.5.1
5. Deleted the entire Libation folder in my User folder. I had to rescan my library for Libation to know which audiobooks have already been downloaded.

**Platform**

MacOS 12.7 on MacBook Air, 2017

**Log Files**
[_Log202406.zip](https://github.com/user-attachments/files/15808720/_Log202406.zip)

**Comment by rmcrackan** [comment]
Thank you so much for the thorough explanation and for the log file. Can you try [version 11.3.13](https://github.com/rmcrackan/Libation/releases/tag/v11.3.13) and send me the new log file if you're still getting this bug?

**Comment by rmcrackan** [comment]
Thanks. I see one download (Journey to the West - Chapter 14 Part 1 [B0D4DYW7HL].mp3) and no error seems to follow this. Did you still get the crash?

**Comment by VicGar87** [comment]
Yes, the program did crash. I made one more attempt, and it had a slightly different behavior than before in that it stalled for about 7 minutes at the end of the download before crashing. I took a screenshot video of it this time, and saved the operating system crash report.

[_Log202406 (11.3.13) 6-13-24.zip](https://github.com/user-attachments/files/15824039/_Log202406.11.3.13.6-13-24.zip)
[Libation crash report  6-13-24.txt](https://github.com/user-attachments/files/15824064/Libation.crash.report.6-13-24.txt)

https://github.com/rmcrackan/Libation/assets/172528461/7b56cf18-473c-4a53-9e8e-b8ed77a5a66d

**Comment by rmcrackan** [comment]
Wow -- thank you for all the effort you've put in. Unfortunately, nothing here points to a specific error.

> This behavior started approximately one month ago

Did something about your operating system change a month ago? The part my brain keeps going back to is how it used to work. I'm thinking, maybe for testing's sake, what if we reset everything. If you rename (not delete) your the Libation folder (inside your users directory), then Libation can't find it and it will start as though it's brand new. If we get this working again, I can help migrate your old database info into the new one if needed.

So, you rename that folder and restart Libation. Before re-adding your account, let's turn off auto-scan (Import -> Auto Scan Library). This will also help cut down on variables with which we're contending. Then you can add your account and do a manual scan. At this point it's not trying to download anything. Find 1 book which you haven't downloaded before and click its stoplight. Then let it rest for a few minutes to see if it causes your previous behavior again.

**Comment by VicGar87** [comment]
I wondered as well if something on my computer changed to make it start doing this. I ran a system update, and my operating system is now 12.7.5. I did as you instructed, renamed the Libation folder, and the software acted as if i'm a new user. I disabled auto-scan, added my user, manually scanned the library, and clicked the traffic light to download. Alas, it has acted the same: downloaded, then crashed right when it finishes. This is the beginning of the crash report:

Process:               Libation [2073]
Path:                  /Volumes/VOLUME/*/Libation.app/Contents/MacOS/Libation
Identifier:            org.libation.macos
Version:               11.3.13
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-06-13 14:04:17.3327 -0500
OS Version:            macOS 12.7.5 (21H1222)
Report Version:        12
Anonymous UUID:        B583CFEF-58D9-5108-B9C7-F231A71FC5D8

[_Log202406 (OS 12.7.5 - Libation 11.3.13 - 6-13-24).zip](https://github.com/user-attachments/files/15827585/_Log202406.OS.12.7.5.-.Libation.11.3.13.-.6-13-24.zip)

**Comment by VicGar87** [comment]
If this seems to be a rare issue that takes more time than is beneficial, I may just resign myself to downloading one at a time at a casual pace. I appreciate all the time you've spent thus far helping me. Hopefully this issue can be resolved in the future though. If you can think of anything else to try, I'm open to it.

**Comment by rmcrackan** [comment]
I feel awful having to do this, but I'm going to close the issue. Not because it's solved but because I'm out of ideas. You've been the absolute gold standard of a bug reporter and I simply cannot find a single viable thread to pull on.

* Your log files have no errors which means something is shutting down Libation before letting it write to the log.
* It used to work yet a fresh install didn't help.
* Your login worked fine.
* I see evidence of it reading and writing correctly to the database so presumably that's not an issue.
* I see a successful scan and successful download so communication with audible works.
* It's writing non-errors to the log so the logging itself isn't busted.
* It will download a single book fine and evidently move it to the correct place.
* I've heard of certain json config files getting corrupted. That's solved though by just deleting it and letting Libation re-create it. In your case though, you started from scratch so you effectively did that with every file.

I got nothing.

If you know any command line scripting/automation, you could try [LibationCLI](https://github.com/rmcrackan/Libation/blob/master/Documentation/Advanced.md#command-line-interface).

If you think of anything, I still get notifications on closed tickets. Sorry I couldn't crack this one :(

**Comment by rmcrackan** [comment]
@VicGar87 Thank you for the follow up. This is such a relief to hear!!

**Comment by VicGar87** [comment]
well, I thought perhaps I spoke too soon, because right after I updated this thread, I had a crash. But interestingly, I think I pinpointed where the bug is, and I am able to circumvent it with a workaround.
When I have podcasts visible in the library list, when I download a book, the software crashes when it completes the download of an audiobook or podcast episode. But when I hide podcasts using "-podcast"  in the filter box, the software works correctly without crashing. I can even download podcasts, but quickly hide them with a filter, and it will download them without issue. Apparently having podcasts visible in the library list makes the software crash when it updates the library list to mark any download as completed, as far as I can tell
